### PR TITLE
WIP avoid dirty suffix in release tag

### PR DESCRIPTION
**Pull request by ctreatma**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -82,8 +82,6 @@ jobs:
             conventional-changelog-conventionalcommits@4.6.3
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - name: Set PreRelease Version
-        run: echo "GORELEASER_CURRENT_TAG=v$(pulumictl get version --language generic)" >> $GITHUB_ENV
```
Comment (ctreatma) [inline .github/workflows/release.yml:86]: This is _not_ the solution, but I made this change as a starting point for discussion.

What seems to be happening here is that the release job explicitly runs `make tfgen`, which produces changes to the file `provider/cmd/pulumi-resource-equinix/schema.json`.

I'm not clear why these changes happen; I can see that that file was updated in the upgrade-provider PR #139.  Do we need to run extra commands as part of the upgrade-provider action?  Is that something we _can_ do?

We could update the `.releaserc.json` so that the semantic-release action commits any changes it finds, but that feels wrong (the semantic release action is meant to only commit things that are specifically related to the version number change, not things that were missed by earlier actions).

Comment (ocobles) [reply .github/workflows/release.yml:86]: I didn't see this happen during the release job, although it had happened to me before locally. This has to do with the process to generate the examples. The upgrade-provider command uses `make tfgen`, so that was the only place I was able to add `generate_examples`, the problem was that `generate_examples` needs the plugin to be installed so that was creating a circular dependency between `make tfgen` `make provider` and `make generate_examples` which I solved with some workarounds. I think that under some circumstances that I am not aware of right now, that fix is ​​failing

Comment (ocobles) [reply .github/workflows/release.yml:86]: nvm I just saw the release actions error https://github.com/equinix/pulumi-equinix/actions/runs/10393580883/job/28782845874

Comment (ctreatma) [reply .github/workflows/release.yml:86]: I was wrong about the task run in the release workflow, it's [`make only_tfgen`](https://github.com/equinix/pulumi-equinix/pull/152/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R61).  When I run that task locally on the latest commit from `main`, I see changes to the schema file.  If `make tfgen` is already running during provider upgrade, does that mean we also have to run it after one or more of the other upgrade PRs to ensure that the codebase is up-to-date before release?

Comment (ocobles) [reply .github/workflows/release.yml:86]: In [#140](https://github.com/equinix/pulumi-equinix/pull/140) manual changes were introduced on the examples, that's why `make tfgen` in goreleaser produces that diff. So we need a couple of changes - First, fix the bug that was causing the examples to show dirty in the go.mod versions, and second, replace make tfgen with only_tfgen in goreleaser. We can also reinforce the message of avoiding manual changes in the examples

Comment (ocobles) [reply .github/workflows/release.yml:86]: I'm going to open a new PR with the required changes in the examples/schema. Do you want to update this PR to replace `make only_tfgen` in the .goreleaser.yml file?

Comment (ctreatma) [reply .github/workflows/release.yml:86]: #140 hasn't been merged yet.  The uncommitted changes in the release job are unrelated to that PR

File: provider/cmd/pulumi-resource-equinix/schema.json [hunk 2]
```diff
@@ -12810,7 +12810,7 @@
...
...()\n                .purchaseOrderNumber(\"1-323292\")\n                .build())\n            .location(CloudRouterLocationArgs.builder()\n                .metroCode(\"SV\")\n                .build())\n            .package_(CloudRouterPackageArgs.builder()\n                .code(\"STANDARD\")\n                .build())\n            .project(CloudRouterProjectArgs.builder()\n                .projectId(\"776847000642406\")\n                .build())\n            .account(CloudRouterAccountArgs.builder()\n                .accountNumber(\"203612\")\n                .build())\n            .build());\n\n    }\n}\n```\n```yaml\nresources:\n  newCloudRouter:\n    type: equinix:fabric:CloudRouter\n    name: new_cloud_router\n    properties:\n      name: Router-SV\n      type: XF_ROUTER\n      notifications:\n        - type: ALL\n          emails:\n            - example@equinix.com\n            - test1@equinix.com\n      order:\n        purchaseOrderNumber: 1-323292\n      location:\n        metroCode: SV\n      package:\n        code: STANDARD\n      project:\n        projectId: '776847000642406'\n      account:\n        accountNumber: '203612'\n```\n{{% /example %}}\n\n{{% /examples %}}",
```
Comment (ctreatma) [inline provider/cmd/pulumi-resource-equinix/schema.json:12813]: These are the uncommitted changes that are causing the release job to fail.  I don't know why `make only_tfgen` produces these changes.

The diff is hard to parse visually so I also can't tell what exactly has changed about these descriptions.

Comment (ocobles) [reply provider/cmd/pulumi-resource-equinix/schema.json:12813]: because of this https://github.com/equinix/pulumi-equinix/pull/152#discussion_r1722135351
#140 those manual changes in the examples. The code of the examples are added to the each description once generated

File: provider/cmd/pulumi-resource-equinix/schema.json [hunk 3]
```diff
@@ -12810,7 +12810,7 @@
...
```
Comment (ctreatma) [inline provider/cmd/pulumi-resource-equinix/schema.json:12813]: In this case it appears the change is that the old description had examples that referenced `CloudRouterNotificationArgs`, but the new description has some sort of generic object definition `{ "key": "value" ... }` instead.  Why did `make tfgen` produce references to CloudRouterNotificationArgs but `make only_tfgen` doesn't?

File: provider/cmd/pulumi-resource-equinix/schema.json (hunk 2 above)
Comment (ctreatma) [reply provider/cmd/pulumi-resource-equinix/schema.json:12813]: I don't understand why you're referencing #140.  It hasn't been merged so why would the changes in that unmerged PR affect the release job that runs against the `main` branch?

File: provider/cmd/pulumi-resource-equinix/schema.json (hunk 3 above)
Comment (ctreatma) [reply provider/cmd/pulumi-resource-equinix/schema.json:12813]: Currently the `tfgen` task is defined as:

```make
tfgen: only_tfgen generate_examples
```

If `only_tfgen` updates the schema, and that schema includes examples that are changed by `generate_examples`, perhaps we need to redefine the `tfgen` task so that it runs `only_tfgen` to update the schema JSON so that `generate_examples` generates the correct examples and then runs `only_tfgen` after the examples are generated to make sure any changes to the examples are reflected in the schema JSON.

File: provider/cmd/pulumi-resource-equinix/schema.json (hunk 2 above)
Comment (ocobles) [reply provider/cmd/pulumi-resource-equinix/schema.json:12813]: oops sorry! I don't know how I could have gotten confused, I thought 140 was already merged

File: provider/cmd/pulumi-resource-equinix/schema.json (hunk 3 above)
Comment (ocobles) [reply provider/cmd/pulumi-resource-equinix/schema.json:12813]: I tried adding just`build_schema` after `examples` and it works well without dirty versions

`generate_examples: examples build_schema`

**Comment by ctreatma** [comment]
We figured out what we needed to and made fixes in separate PRs.  Closing as there's nothing in this PR that needs to stick around.
